core/plugins/org.polarsys.capella.core.data.migration/src/org/polarsys/capella/core/data/migration/contribution/ImagePathInRichTextAttributeContribution.java

```diff
@@ -275,6 +279,38 @@ private Map<String, IFile> createFiles(EObject notifier, EAttribute attribute, S
     return createdFiles;
   }
 
+  private IFolder getExistingImagesFolder(EObject notifier) {
+    IFolder imageFolder = null;
+    IProject project = EcoreUtil2.getProject(notifier);
+    try {
+      IResource[] members = project.members();
+      List<IFolder> imagesFolders = Arrays.stream(members)
+              .filter(IFolder.class::isInstance)
+              .map(IFolder.class::cast)
+              .filter(folder -> ImageManager.IMAGE_FOLDER_NAME.equals(folder.getName().toLowerCase()))
+              .toList();
+      if (imagesFolders.size() == 1) {
+        imageFolder = imagesFolders.get(0);
+      } else if (imagesFolders.size() > 1) {
+        Optional<IFolder> optImagesFolder = imagesFolders.stream().filter(folder -> ImageManager.IMAGE_FOLDER_NAME.equals(folder.getName())).findFirst();
+        if (optImagesFolder.isPresent()) {
+          imageFolder = optImagesFolder.get();
+        } else {
+          imageFolder = imagesFolders.get(0);
+        }
+      }
+    } catch (CoreException e) {
+      Activator.getDefault().getLog()
+      .error(MessageFormat.format(Messages.MigrationAction_Image_BrowseProjectToFindExistingImagesProject, project.getName()), e);
+    }
+    if (imageFolder == null) {
+      imageFolder = EcoreUtil2.getProject(notifier).getFolder(ImageManager.IMAGE_FOLDER_NAME);
```

Comment:
very complicated to understand 
I propose
IFolder imageFolder = EcoreUtil2.getProject(notifier).getFolder(ImageManager.IMAGE_FOLDER_NAME);
if (imageFolder == null) {
Arrays.stream(project.members())
              .filter(IFolder.class::isInstance)
              .map(IFolder.class::cast)
              .filter(folder -> ImageManager.IMAGE_FOLDER_NAME.equals(folder.getName().toLowerCase()))
              .findFirst()

Reply:
`org.eclipse.core.internal.resources.Container.getFolder(String)` seems to create a new folder rather than trying to retrieve an existing one, so the result of ` EcoreUtil2.getProject(notifier).getFolder(ImageManager.IMAGE_FOLDER_NAME)` will never be null.